Docs 7996 add kerberos example to HDFS (#746)
mulesoft/docs-connectors: modules/ROOT/pages/hdfs/hdfs-connector-studio.adoc

= Use Studio to Configure Hadoop (HDFS) Connector - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

Anypoint Studio (Studio) editors help you design and update your Mule applications, properties, and configuration files.

> To configure a connector in Studio:

Reviewer: Looks more like "To prepare for configuring a connector in Studio:", with the final step being the actual configuration....

. Create a Mule project.
. Supply your authentication credentials to Anypoint Platform.
. Add the connector to your Mule project
. Configure an input source for the connector's flow.
. Configure the connector.

> Follow these best practices when you configure an input source and connector in Studio:

Reviewer: These belong in the steps/subtask texts to which they refer.

> * Configure an associated global element for the input source and connector.

Reviewer: This is a peer topic below, as if one of the steps above.

> * Create a YAML or properties file to contain your properties, and then reference the file from the configuration.

Reviewer: This is not part of this topic set?

> * Configure a reconnection strategy for the input source and target connector.

Reviewer: Not part of the topic set?

Reviewer: Summary of topic set org:

= Use Studio to Configure Hadoop (HDFS) Connector - Mule 4
==   Add the Connector to Your Project
==   Configure an Input Source
==   Configure a Global Element for the Connector
==   Configure Authentication
===    Kerberos
===    Simple

Reviewer: Compare these actual subtopics to the outline provided in this intro; they don't obviously match or map?

> == Add the Connector to Your Project

Reviewer: This is Step 3 above and still prerequisite to the topic goal, Step 5?
This topic set leads up to configuring the connector, not actually doing it....
```suggestion
== Add the Connector to Your Mule Project
```
Is this later referred to as "your application"?

> Add Hadoop (HDFS) Connector to a Mule project to automatically populate the XML code with the connector's namespace and schema location, as well as add the required dependencies to the project's `pom.xml` file.

Reviewer: ```suggestion
Add Hadoop (HDFS) Connector to your Mule project to automatically populate the project's XML code with the connector's namespace and schema location, as well as to add the required dependencies to the project's `pom.xml` file.
```

> . In Studio, create a Mule project.

Reviewer: This was already done in Step 1 at the top, so it is not really part of ADDING the connector to the project (which already exists).

> . In the Mule Palette view, click *(X) Search in Exchange*.

Reviewer: See suggested replacement above.

> . In *Add Modules to Project*, type HDFS in the search field.

Reviewer: ```suggestion
. In *Add Modules to Project*, type `HDFS` in the search field.
```
Wasn't this same step "type `hdfs` in the search field" earlier? Input can be either uc or lc, right?

> . Click the Hadoop (HDFS) connector in *Available modules*.

Reviewer: ```suggestion
. Click *Hadoop (HDFS) connector* in *Available modules*.
```
Quote the label of that which is clicked, or state this more generally, like "Click the name of the connector in *Available modules*.

. Click *Add*.
. Click *Finish*.

> When you design your application in Studio, the act of dragging the connector's operation from the palette onto the Anypoint Studio canvas automatically populates the XML code with the connector namespace and schema location.

Reviewer: A version of this sentence already introduces the task steps. 

Why do we repeat it here, and how does it relate to the steps above it:

* Does "design your application" refer to creating the Mule project?
If so, avoid changing the wording.
* When is the connector "dragged" to the canvas?
Isn't adding it enough?
* The XML code of your application/project?

> Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.

Reviewer: This looks like a more likely sentence with which to precede the task steps.

> == Configure an Input Source

Reviewer: This is Step 4 leading up to the eventual configuration of the connector?
```suggestion
== Configure an Input Source to Initiate Connector Flow
```
...although it looks as if the only input source that needs configuration is the Read operation, so is this title misleading?

Reviewer: Maybe this is really == Choose an Input Source to Initiate Connector Flow

An input source initiates a flow when a specified condition is met. Possible input source operations for this connector are:

* Read +

> Start your app when the content of a file is read and streamed to the rest of the flow.

Reviewer: ```suggestion
Starts your app when the content of a file is read and streamed to the rest of the flow.
If you use this source, you must provide these values:
** Configuration....
** Path....
```

* HTTP Listener +

> Start your app using a browser or a command such as `curl`.

Reviewer: ```suggestion
Starts your app using a browser or a command such as `curl`.
```

* Scheduler +

> Start your app at timed intervals.

Reviewer: ```suggestion
Starts your app at timed intervals.
```

In Studio, drag the input operation you choose from the *Mule Palette* to the *Source* panel in the canvas.

The required fields for the `Read` input source are:

* Configuration +
The name of the connector configuration to use
* Path +

> The path for the file to read 

Reviewer: See suggestion above.

All of the text in this topic is just context for a single step: dragging an input source operation from palette to canvas.

Unless you want to make this task topic two steps:
. Choose an input source.
. Drag it to the canvas.

I also find the wording around "an input source" vs. "an input source operation" a bit confusing.

[[configure_global_element]]

> == Configure a Global Element for the Connector

Reviewer: Again, this is prerequisite to actually configuring the connector, the subject of this topic set.
Also, this is not listed as a step in the beginning of the topic set, but rather as a "best practice," outside of the initial steplist; it should be between the input source step and the connector configuration step (which is supposed to be the whole task, according to the container topic title).

Author: Not a pre-req It's totally optional and only a best practice. However, this is part of configuring a connector. You can't configure a global element without a connector. It's only one aspect of configuring a connector, but it is nevertheless configuring the connector.

> To configure a connector in your Mule app, first configure a global element that can be used by all instances of that connector in the flow:

Reviewer: "To configure a connector in your Mule app, first configure a global element" 
The way that this is worded, we have not yet begun to configure the connector (the goal of the topic set), and THIS topic is prerequisite to that.
Again, I feel like I have been reading about how to prepare, in Studio, to configure a connector, not the actual configuration of the connector.

> . In the Mule Palette view, search for the connector and select an operation.

Reviewer: ```suggestion
. In the *Mule Palette* view, search for the connector and select an operation.
```
An input source operation, or some other kind of operation?

> . Drag the operation onto the canvas to the right of the input source. 

Reviewer: Didn't we do this already, as the single step of the previous task topic?

Author: No, that was an input source (the thing that starts the flow). This is an operation (2nd part of flow)

> . In the configuration screen for the operation, click the plus sign (+) next to the *Connector configuration* field to add a global element for the connector.

Reviewer: Is this also where the user would further configure the Read operation, if that is the source chosen?

> . In the *General* tab, enter a name for the configuration, and select the connection type from *Connection*. +
> Supported connection types for this connector are: 
> * *Kerberos*
> * *Simple*
> +
> For information about how to configure the connections, see <<configure_authentication,Configure Authentication>>.

Reviewer: Hmm...I see what you did here and below.
This step should really be...:
```suggestion
. In the *General* tab, enter a name for the configuration, select either the Kerberos or Simple connection type from *Connection,* and complete the dialog box provided to configure that connection type.+
```
What would logically follow here are the substeps/values for configuring each type...but that would hugely interrupt this task. 
But as part of this task, shouldn't we present configure Kerberos and configure Simple as subtasks, instead of creating the container of Configure Authorization to hold them?

> . Click the *Advanced* tab to specify a reconnection strategy, if supported.

Reviewer: In the best practices list.
As suggested above, maybe the best practices should be part of their tasks and not separately called out in the intro.

. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

> 

Reviewer: And here ENDS the set of tasks leading to/comprising using Studio to configure this connector?
Is configuring the global element = configuring the connector?

[[configure_authentication]]

> == Configure Authentication

Reviewer: I think what follows is actually supplementary Reference information, not even subtasks to configuring the global element: Fields Used to Configure Kerberos Connection Authentication and Fields Used to Configure Simple....

Reviewer: Tables = Reference info
We really should not word field descriptions as user actions.
Maybe something like "Complete the fields of the connection type configuration dialog box in the global element configuration task as follows:"?

Reviewer: BREAK HERE TO TALK WITH VALKYRIE

> To access the data in a Hadoop (HDFS) instance, you must authenticate your application’s requests using a supported authentication method. You can select the authentication method when you <<configure_global_element,configure a global element>> for the connector. 

Reviewer: ```suggestion
To access the data in a Hadoop (HDFS) instance, you must authenticate your application’s requests using the authentication method you chose when you <<configure_global_element,configured a global element>> for the connector, either Kerberos or Simple. 
```

For Hadoop (HDFS) Connector, the following connection authentication types are supported:

* <<kerberos,Kerberos>>
* <<hdfs_simple,Simple>>

> 

Reviewer: See suggestion above
